code/game/gamemodes/clock_cult/clock_effects/clock_sigils.dm

```diff
@@ -314,21 +314,28 @@
 					revival_cost = 0
 				var/mob/dead/observer/ghost = L.get_ghost(TRUE)
 				if(GLOB.clockwork_vitality >= revival_cost && (ghost || (L.mind && L.mind.active)))
-					if(ghost)
-						ghost.reenter_corpse()
-					L.revive(1, 1)
-					var/obj/effect/temp_visual/ratvar/sigil/vitality/V = new /obj/effect/temp_visual/ratvar/sigil/vitality(get_turf(src))
-					animate(V, alpha = 0, transform = matrix()*2, time = 8)
-					playsound(L, 'sound/magic/staff_healing.ogg', 50, 1)
-					L.visible_message("<span class='warning'>[L] suddenly gets back up, [L.p_their()] body dripping blue ichor!</span>", "<span class='inathneq'>\"[text2ratvar("You will be okay, child.")]\"</span>")
-					GLOB.clockwork_vitality -= revival_cost
+					if(L.has_status_effect(STATUS_EFFECT_ICHORIAL_STAIN))
+						visible_message("<span class='boldwarning'>[src] strains, but nothing happens...</span>")
+						if(L.pulledby)
+							to_chat(L.pulledby, "<span class='userdanger'>[L] was already revived recently by a vitality matrix! Wait a bit longer!</span>")
+						break
+					else
+						if(ghost)
+							ghost.reenter_corpse()
+						L.revive(1, 1)
+						var/obj/effect/temp_visual/ratvar/sigil/vitality/V = new /obj/effect/temp_visual/ratvar/sigil/vitality(get_turf(src))
+						animate(V, alpha = 0, transform = matrix()*2, time = 8)
+						playsound(L, 'sound/magic/staff_healing.ogg', 50, 1)
+						L.visible_message("<span class='warning'>[L] suddenly gets back up, [L.p_their()] body dripping blue ichor!</span>", "<span class='inathneq'>\"[text2ratvar("You will be okay, child.")]\"</span>")
```

Comment:
You can move the `<span class='warning'>[L] suddenly gets back up, [L.p_their()] body dripping blue ichor!</span>` to the status effect and just make the `<span class='inathneq'>\"[text2ratvar("You will be okay, child.")]\"</span>` a to_chat()